Content.Shared/Storage/Components/SharedMapLayerData.cs

```diff
@@ -19,8 +19,8 @@ public sealed class SharedMapLayerData
     {
         public string Layer = string.Empty;
 
-        [DataField("whitelist", required: true)]
-        public EntityWhitelist Whitelist { get; set; } = new();
+        [DataField("whitelist", required: true, serverOnly: true)]
+        public EntityWhitelist ServerWhitelist { get; set; } = new();
```

Comment:
While this is in shared, the whitelist is only ever used in server-side code. Currently it includes server-exclusive components and can cause issues if the client were to ever use this whitelist.

Eventually the layer-mapper stuff should become a client-side visualizer system, and the server-exclusive components either need to be shared or moved to tags.